src/BaseVault.sol

```diff
@@ -775,11 +781,44 @@ abstract contract BaseVault is IVault, ERC20PermitUpgradeable, AccessControlUpgr
     }
 
     function _processAccounting() internal virtual {
+        uint256 exchangeRateBefore = IERC4626(address(this)).convertToAssets(10 ** decimals());
+
         uint256 totalBaseBalance = computeTotalAssets();
 
         _getVaultStorage().totalAssets = totalBaseBalance;
         // solhint-disable-next-line not-rely-on-time
         emit ProcessAccounting(block.timestamp, totalBaseBalance);
+
+        address hooks_ = address(hooks());
```

Comment:
nitpick, but the hooks() can be cast to an address just inside the if

so you can Read

```IHooks hooks_ = hooks();```